### PR TITLE
Fixed #26687 -- Made i18n test not use hardcoded path separator.

### PR DESCRIPTION
**Pull request by ramiro**
This fails on Windows when the test run time conditions are such that
the assertion is really executed.

### CODE REVIEW

File: tests/i18n/test_extraction.py [hunk 1]
```diff
@@ -534,27 +534,25 @@ def tearDown(self):
...
+        if os.path.exists(self.symlinked_dir):
+            self.assertTrue(os.path.islink(self.symlinked_dir))
+        else:
+            # On Python >= 3.2) os.symlink() exists always but then can
+            # fail at runtime when user hasn't the needed permissions on
+            # Windows versions that support symbolink links (>= 6/Vista).
+            # See Python issue 9333 (http://bugs.python.org/issue9333).
+            # Skip the test in that case
+            try:
+                os.symlink(os.path.join(self.test_dir, 'templates'), self.symlinked_dir)
```
Comment (timgraham) [inline tests/i18n/test_extraction.py]: I'm confused -- does Python 2 never reach this branch?

Comment (ramiro) [inline tests/i18n/test_extraction.py]: My bad. I forgot completely about Python 2 and broke it on Windows. I've undone that extra bogus change on a new commit.

**Comment by timgraham** [comment]
buildbot, test this please.

**Comment by timgraham** [comment]
I've installed gettext on the Windows machine and saw this test fail on another build. Now it's passing although there's still one more unrelated failure to solve separately to get the build green.

**Pull request merged**